packages/jest-worker/src/Farm.ts

```diff
@@ -44,41 +46,66 @@ export default class Farm {
     }
   }
 
-  doWork(method: string, ...args: Array<any>): Promise<unknown> {
-    return new Promise((resolve, reject) => {
-      const computeWorkerKey = this._computeWorkerKey;
-      const request: ChildMessage = [CHILD_MESSAGE_CALL, false, method, args];
+  doWork(
+    method: string,
+    ...args: Array<any>
+  ): PromiseWithCustomMessage<unknown> {
+    let customMessageListeners: Array<OnCustomMessage> = [];
+
+    const addCustomMessageListener = (listener: OnCustomMessage) => {
+      customMessageListeners.push(listener);
+      return () => {
+        customMessageListeners = customMessageListeners.filter(
+          l => l !== listener,
+        );
+      };
+    };
+
+    const onCustomMessage: OnCustomMessage = (message: any) => {
```

Comment:
```suggestion
    const onCustomMessage: OnCustomMessage = message => {
```

`OnCustomMessage` should make sure this can be inferred, no?